libsolutil/CommonData.h

```diff
@@ -38,39 +38,45 @@
 /// Operators need to stay in the global namespace.
 
 /// Concatenate the contents of a container onto a vector
-template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U const& _b)
+template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U& _b)
 {
 	for (auto const& i: _b)
 		_a.push_back(i);
 	return _a;
 }
 /// Concatenate the contents of a container onto a vector, move variant.
-template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U&& _b)
+template <class T, class U,
+	class = std::enable_if_t<!std::is_lvalue_reference_v<U>>>
```

Comment:
Just to be clear: With the change above you really don't need this change anymore now :-).
If you want, you can just move the condition into a static assertion in the body - but lvalues will always choose the version above now, so changing nothing in this version would actually be fine now.
(and of course the same holds for the other cases)

Reply:
ah really? will check that.. compiling on my machine takes sooooo looong :/